### DISCUSSION TITLE
Different set of lints for tests?

### DISCUSSION THREAD

**Discussion by nbigaouette** [rust-lang/rust-clippy#11343] [Q&A]
I enable a lot of lints for my code, for example I use `-Wclippy::pedantic -Dclippy::unwrap_used` for clippy settings in VSCode's  `settings.json`'s `rust-analyzer.check.extraArgs`.

This way I can stay in my editor and get aggressive feedback for things that are enforced in CI.

The issue I have is that those lints are _too aggressive_ for tests; I'm fine with unwrapping in unit test for example.

Is there a way to have two sets of lints (at the command line), one for the main code and the other for `#[cfg(test)]`?

Thanks!

**Comment by flip1995** [comment]
If you run `cargo clippy -- -Wclippy::pedantic` it should not trigger on tests. However if you run `cargo clippy --all-targets -- -Wclippy::pedantic`, it will also run on tests. So for a separate set of lints on the command line for tests, you can try `cargo clippy --tests -- <set-of-lints-flags>`. Not sure how to best configure rust-analyzer to not apply all of the lints in `extraArgs` for tests.

In general it is probably best to define lints in the crate as inner attributes, instead of the command line. Then everyone working on the crate has the same set of lints enabled, and it doesn't depend on their r-a config.
